pkg/logicalbackup/logicalbackup.go

```diff
@@ -172,109 +160,121 @@ func New(ctx context.Context, stopCh chan struct{}, cfg *config.Config) (*Logica
 	lb := &LogicalBackup{
 		ctx:    ctx,
 		stopCh: stopCh,
-		dbCfg:  pgxConn,
-
-		replMessageWaitTimeout: waitTimeout,
-		statusTimeout:          statusTimeout,
-		backupTables:           NewBackupTables(),
-		relationMessages:       make(map[dbutils.Oid][]byte),
-		tableNameChanges:       oidToName{nameChangeHistory: make(map[dbutils.Oid][]NameAtLsn)},
-		pluginArgs:             []string{`"proto_version" '1'`, fmt.Sprintf(`"publication_names" '%s'`, cfg.PublicationName)},
-		basebackupQueue:        queue.New(ctx),
-		waitGr:                 &sync.WaitGroup{},
-		cfg:                    cfg,
-		msgCnt:                 make(map[cmdType]int),
+
+		backupTables:     NewBackupTables(),
+		relationMessages: make(map[dbutils.OID][]byte),
+		tableNameChanges: oidToName{nameChangeHistory: make(map[dbutils.OID][]NameAtLSN)},
+		pluginArgs:       []string{`"proto_version" '1'`, fmt.Sprintf(`"publication_names" '%s'`, cfg.PublicationName)},
+		basebackupQueue:  queue.New(ctx),
+		waitGr:           &sync.WaitGroup{},
+		cfg:              cfg,
+		msgCnt:           make(map[cmdType]int),
 		srv: http.Server{
-			Addr:    fmt.Sprintf(":%d", 8080),                     // TODO: get rid of the hardcoded value
-			Handler: http.TimeoutHandler(mux, time.Second*20, ""), // TODO: get rid of the hardcoded value
+			Addr:    fmt.Sprintf(":%d", httpSrvPort),
+			Handler: http.TimeoutHandler(mux, httpSrvTimeout, ""),
 		},
 		prom: prom.New(cfg.PrometheusPort),
 	}
 
-	if cfg.StagingDir != "" {
-		if _, err := os.Stat(cfg.StagingDir); os.IsNotExist(err) {
-			if err := os.Mkdir(cfg.StagingDir, os.ModePerm); err != nil {
-				return nil, fmt.Errorf("could not create staging dir: %v", err)
+	if err := createDirs(cfg.StagingDir, cfg.ArchiveDir); err != nil {
+		return nil, err
+	}
+
+	if err := lb.initDB(); err != nil {
+		return nil, err
+	}
+
+	if err := lb.registerMetrics(); err != nil {
+		return nil, err
+	}
+
+	return lb, nil
+}
+
+func createDirs(dirs ...string) error {
+	for _, dir := range dirs {
+		if dir == "" {
+			continue
+		}
+
+		if _, err := os.Stat(dir); os.IsNotExist(err) {
+			if err := os.Mkdir(dir, os.ModePerm); err != nil {
+				return fmt.Errorf("could not create %q dir: %v", dir, err)
 			}
 		} else if err != nil {
-			return nil, fmt.Errorf("%q stat error: %v", cfg.StagingDir, err)
+			return fmt.Errorf("%q stat error: %v", dir, err)
 		}
 	}
 
-	if _, err := os.Stat(cfg.ArchiveDir); os.IsNotExist(err) {
-		if err := os.Mkdir(cfg.ArchiveDir, os.ModePerm); err != nil {
-			return nil, fmt.Errorf("could not create archive dir: %v", err)
-		}
-	} else if err != nil {
-		return nil, fmt.Errorf("%q stat error: %v", cfg.ArchiveDir, err)
-	}
+	return nil
+}
+
+func (b *LogicalBackup) initDB() error {
```

Comment:
The naming clash with the Postgres tool is unfortunate; perhaps connectAndPrepareDatabase?

Reply:
I'd name it `prepareDB`